crate/findex/src/adt.rs

```diff
@@ -51,6 +51,32 @@ pub trait VectorADT: Send {
     fn read(&self) -> impl Send + Future<Output = Result<Vec<Self::Value>, Self::Error>>;
 }
 
+/// This trait  extends the functionality of the standard `IndexADT` by
+/// providing methods that operate on multiple keywords or entries
+/// simultaneously.
+#[cfg(feature = "batch")]
+pub trait IndexBatcher<Keyword, Value> {
```

Comment:
You say this is an extension, but this trait is not a super-trait of `IndexADT`.